src/platform/modulesmanager/inc/ModulesManager.h

```diff
@@ -49,59 +44,59 @@ class ModulesManagerLog
     static OSCONFIG_LOG_HANDLE m_log;
 };
 
+class MpiSession;
+
 class ModulesManager
 {
 public:
-    ModulesManager(std::string clientName, const unsigned int maxPayloadSizeBytes = 0);
-    virtual ~ModulesManager();
-
-    std::string GetClientName();
-
-    // Set the default cleanup timespan in seconds for Management Modules (MMs), unloads the MM if Lifespan is "Short"
-    void SetDefaultCleanupTimespan(unsigned int timespan);
+    ModulesManager();
+    ~ModulesManager();
 
     // Loads Management Modules (MMs) into the Modules Manager
-    int LoadModules();
-    int LoadModules(std::string modulePath);
     int LoadModules(std::string modulePath, std::string configJson);
 
-    int MpiSet(const char* componentName, const char* objectName, const MPI_JSON_STRING payload, const int payloadSizeBytes);
-    int MpiGet(const char* componentName, const char* objectName, MPI_JSON_STRING* payload, int* payloadSizeBytes);
-    int MpiSetDesired(const MPI_JSON_STRING payload, const int payloadSizeBytes);
-    int MpiGetReported(MPI_JSON_STRING* payload, int* payloadSizeBytes);
-
     void UnloadModules();
-    void UnloadAllModules();
-    void DoWork();
 
 protected:
-    struct ModuleMetadata
-    {
-        std::shared_ptr<ManagementModule> module;
-        std::chrono::system_clock::time_point lastOperation;
-        bool operationInProgress;
-    };
+    // Component name -> vector of reported components according to osconfig.json
+    std::map<std::string, std::unordered_set<std::string>> m_reported;
 
-    // Module mapping componentName <-> ModuleMetadata
-    std::map<std::string, ModuleMetadata> modMap;
-    std::vector<ModuleMetadata> modulesToUnload;
+    // Component name -> module name
+    std::map<std::string, std::string> m_componentToModule;

```

Comment:
>m_componentToModule [](http://example.com/codeflow?start=39&length=19)

This is a string, a state. Not a change from one state to another. Change to m_moduleComponentName or similar?

Reply:
Agreed, and changed to `m_moduleComponentName`